src/battle_transition.c

```diff
@@ -3908,17 +3911,29 @@ void Task_BattleTransition_Intro(u8 taskId)
 
 static bool8 TransitionIntro_FadeToGray(struct Task *task)
 {
+    u8 paletteNum = IndexOfSpritePaletteTag(TAG_WEATHER_START);
+    u16 index = (paletteNum+16)*16+9; // SHADOW_COLOR_INDEX
```

Comment:
```suggestion
    u16 index = (paletteNum + 16) * 16 + 9; // SHADOW_COLOR_INDEX
```
But this should not use all these magic numbers in the first place

Reply:
I think the non-magic version might be:
```c
u16 index = OBJ_PLTT_ID(paletteNum) + 9; // SHADOW_COLOR_INDEX
```

Reply:
Applied.